admin/test/features/step_definitions/manage_check_forms_steps.rb

```diff
@@ -30,8 +30,7 @@
 end
 
 Then(/^it should be added to the list of forms$/) do
-  sleep 2 # need to address this sleep , test fails randomly with out this
-  expect(manage_check_forms_page.available_checks.rows.count).to eql(@current_form_count + 1)
+  wait_until{expect(manage_check_forms_page.available_checks.rows.count).to eql(@current_form_count + 1)}
```

Comment:
i think if you put an expectation in the wait utility library, it ignores it. you may have to do this:
wait_until{manage_check_forms_page.available_checks.rows.count ==(@current_form_count + 1)}
expect(manage_check_forms_page.available_checks.rows.count).to eql(@current_form_count + 1)
looks repetitive but is robust.
try it once by giving a false condition like 
wait_until{expect(manage_check_forms_page.available_checks.rows.count).to_not eql(@current_form_count + 1)}
if the step still passes with a not condition, then the wait utility gem ignore expectations.